AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -222,6 +222,20 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
         }
     }
 
+    private func processSuccessEvent(_ mutationEvent: MutationEvent,
+                                     mutationSyncMetadata: MutationSync<AnyModel>?) {
+        if let mutationSyncMetadata = mutationSyncMetadata {
+            MutationEvent.updatePendingMutationEventVersionIfNil(
+                for: mutationEvent.modelId,
+                mutationSync: mutationSyncMetadata,
+                storageAdapter: storageAdapter) { result in
+                self.log.verbose("\(#function) received result after syncing version from API: \(result)")
+            }
+        }
+
+        completeProcessingEvent(mutationEvent, mutationSyncMetadata: mutationSyncMetadata)
```

Comment:
this is the same problem as before, by performing an async call like `MutationEvent.updatePendingMutationEventVersionIfNil` followed by another method call like `completeProcessingEvent` means at some point, control will be returned to this method and will execute `completeProcessingEvent` while another thread is performing some async work inside `updatePendingMutationEventVersionIfNil`. so the result is indeterminate in the order things are executed. the simplest code to fix this would look like this:

```swift
if let mutationSyncMetadata = mutationSyncMetadata {
            MutationEvent.updatePendingMutationEventVersionIfNil(
                for: mutationEvent.modelId,
                mutationSync: mutationSyncMetadata,
                storageAdapter: storageAdapter) { result in
                self.log.verbose("\(#function) received result after syncing version from API: \(result)")
 // perfom this in the callback of your async calls, so they are performed one after another
      completeProcessingEvent(mutationEvent, mutationSyncMetadata: mutationSyncMetadata)
            }
        } else {
// only perform this when there is no result to handle
        completeProcessingEvent(mutationEvent, mutationSyncMetadata: mutationSyncMetadata)
}


```